onnxmltools/convert/coreml/operator_converters/TensorToProbabilityMap.py

```diff
@@ -39,7 +41,29 @@ def convert_tensor_to_probability_map(scope, operator, container):
     else:
         raise TypeError('Only neural network classifiers and pipeline classifiers are supported')
 
-    container.add_node('ZipMap', [operator.inputs[0].full_name], [operator.outputs[0].full_name],
+    input_shape = operator.inputs[0].type.shape
+    if len(operator.inputs[0].type.shape) != 2:
+        # Calculate the shape attribute of ONNX Reshape
+        if input_shape[0] != 'None':
+            N = input_shape[0]
+        else:
+            N = -1  # -1 means that this dimension is automatically determined in runtime and unknown in conversion time
+
+        if all(isinstance(i, numbers.Integral) for i in input_shape[1:]):
+            C = 1
+            for i in input_shape[1:]:
+                C *= int(i)
+        else:
+            C = -1  # -1 means that this dimension is automatically determined in runtime and unknown in conversion time
+
+        # ZipMap in ONNX only accepts [C] and [N, C] inputs. In cases of [N, C, 1, 1], we reshape the probability tensor
+        # into [N, C] before feeding it into ZipMap.
+        buffer_name = scope.get_unique_variable_name('buffer')
```

Comment:
assert N and C can't be -1 at the same time?

Reply:
Yes, good point!